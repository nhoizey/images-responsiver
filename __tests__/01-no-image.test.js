'use strict';

const prettier = require('prettier');
const imagesResponsiver = require('../index');

function cleanHtml(html) {
	return prettier.format(html, { parser: 'html' });
}

describe('no image', () => {
	test('malformed HTML without image', () => {
		const content = `<p>Hello<br><br>mister<br /></p>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('keeps intact HTML without image', () => {
		const content = `<!DOCTYPE html><html><body><p>Hello</p></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('keeps intact HTML without image with inline style and custom property', () => {
		const content = `<!DOCTYPE html><html><body><p style="--foo: bar">Hello</p></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('keeps intact HTML without image with HTML entity', () => {
		const content = `<!DOCTYPE html><html><body><p>Hello &copy;</p></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});
});
