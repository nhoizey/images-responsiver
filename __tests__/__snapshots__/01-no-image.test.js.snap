// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no image keeps intact HTML without image 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <p>Hello</p>
  </body>
</html>
"
`;

exports[`no image keeps intact HTML without image with HTML entity 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <p>Hello Â©</p>
  </body>
</html>
"
`;

exports[`no image keeps intact HTML without image with inline style and custom property 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <p style=\\"--foo: bar\\">Hello</p>
  </body>
</html>
"
`;

exports[`no image malformed HTML without image 1`] = `
"<p>Hello<br /><br />mister<br /></p>
"
`;
